csharp/LogicalTypeFactory.cs

```diff
@@ -34,6 +34,76 @@ public virtual (Type physicalType, Type logicalType) GetSystemTypes(ColumnDescri
+                {typeof(ulong?), (Repetition.Optional, LogicalType.Int(64, isSigned: false), PhysicalType.Int64)},
+                {typeof(Int96), (Repetition.Required, LogicalType.None(), PhysicalType.Int96)},
+                {typeof(Int96?), (Repetition.Optional, LogicalType.None(), PhysicalType.Int96)},
+                {typeof(float), (Repetition.Required, LogicalType.None(), PhysicalType.Float)},
+                {typeof(float?), (Repetition.Optional, LogicalType.None(), PhysicalType.Float)},
+                {typeof(double), (Repetition.Required, LogicalType.None(), PhysicalType.Double)},
+                {typeof(double?), (Repetition.Optional, LogicalType.None(), PhysicalType.Double)},
+                {typeof(decimal), (Repetition.Required, null, PhysicalType.FixedLenByteArray)},
+                {typeof(decimal?), (Repetition.Optional, null, PhysicalType.FixedLenByteArray)},
+                {typeof(Guid), (Repetition.Required, LogicalType.Uuid(), PhysicalType.FixedLenByteArray)},
+                {typeof(Guid?), (Repetition.Optional, LogicalType.Uuid(), PhysicalType.FixedLenByteArray)},
+                {typeof(Date), (Repetition.Required, LogicalType.Date(), PhysicalType.Int32)},
+                {typeof(Date?), (Repetition.Optional, LogicalType.Date(), PhysicalType.Int32)},
+                {typeof(DateTime), (Repetition.Required, LogicalType.Timestamp(isAdjustedToUtc: true, timeUnit: TimeUnit.Micros), PhysicalType.Int64)},
+                {typeof(DateTime?), (Repetition.Optional, LogicalType.Timestamp(isAdjustedToUtc: true, timeUnit: TimeUnit.Micros), PhysicalType.Int64)},
+                {typeof(DateTimeNanos), (Repetition.Required, LogicalType.Timestamp(isAdjustedToUtc: true, timeUnit: TimeUnit.Nanos), PhysicalType.Int64)},
+                {typeof(DateTimeNanos?), (Repetition.Optional, LogicalType.Timestamp(isAdjustedToUtc: true, timeUnit: TimeUnit.Nanos), PhysicalType.Int64)},
+                {typeof(TimeSpan), (Repetition.Required, LogicalType.Time(isAdjustedToUtc: true, timeUnit: TimeUnit.Micros), PhysicalType.Int64)},
+                {typeof(TimeSpan?), (Repetition.Optional, LogicalType.Time(isAdjustedToUtc: true, timeUnit: TimeUnit.Micros), PhysicalType.Int64)},
+                {typeof(TimeSpanNanos), (Repetition.Required, LogicalType.Time(isAdjustedToUtc: true, timeUnit: TimeUnit.Nanos), PhysicalType.Int64)},
+                {typeof(TimeSpanNanos?), (Repetition.Optional, LogicalType.Time(isAdjustedToUtc: true, timeUnit: TimeUnit.Nanos), PhysicalType.Int64)},
+                {typeof(string), (Repetition.Optional, LogicalType.String(), PhysicalType.ByteArray)},
+                {typeof(byte[]), (Repetition.Optional, LogicalType.None(), PhysicalType.ByteArray)}
+            };
+
+        /// <summary>
+        /// Query whether the given C# type is supported and a schema node can potentially be created.
+        /// </summary>
+        public static bool IsSupported(Type type)
```

Comment:
Suggest a non-static `virtual` method instead (then user-side it can be called on `LogicalTypeFactory.Default` where you need it, or on the `LogicalTypeFactory` instance of the `ParquetFileReader` if the latter is given).

That way you don't need the static dictionary above either.

Reply:
fair enough do you envision the dictionary being passed as a constructor arg instead?

Reply:
`LogicalTypeFactory` already has a constructor taking a dictionary in. Hence why you don't need to worry about it.